HerPublicWebsite/Views/Questionnaire/Ineligible.cshtml

```diff
@@ -62,31 +62,41 @@
 
 <div class="govuk-grid-row">
     <div class="govuk-grid-column-two-thirds">
-        
+
         <h1 class="govuk-heading-l">We do not think you are eligible</h1>
-        
+
         <p class="govuk-body">
+            We do not think you are eligible for the Home Upgrade Grant
+            @{
+                if (Model.ShowWarmHomesText)
+                {
+                    <text>
+                        or the Warm Homes: Local Grant
+                    </text>
+                }
+            }
             @{
-                switch (Model.EpcIsTooHigh, Model.IncomeIsTooHigh)  {
+                switch (Model.EpcIsTooHigh, Model.IncomeIsTooHigh)
+                {
                     case (true, true):
                         <text>
-                            We do not think you are eligible for the Home Upgrade Grant because your EPC rating is A, B, or C and your income is more than £36,000 but you may be able to get other help for your energy needs.
+                            because your EPC rating is A, B, or C and your income is more than £36,000 but you may be able to get other help for your energy needs.
                         </text>
                         break;
                     case (true, false):
                         <text>
-                            We do not think you are eligible for the Home Upgrade Grant because your EPC rating is A, B, or C but you may be able to get other help for your energy needs.
+                            because your EPC rating is A, B, or C but you may be able to get other help for your energy needs.
                         </text>
                         break;
                     default:
                         <text>
-                            We do not think you are eligible for the Home Upgrade Grant because your income is more than £36,000 but you may be able to get other help for your energy needs.
+                            because your income is more than £36,000 but you may be able to get other help for your energy needs.
                         </text>
                         break;
-                };
+                }
             }
         </p>
-        
+
```

Comment:
Something to think about here - this is getting to be quite a lot of business logic in the view layer, which isn't the place for it and it impacts readability a bit. Don't think you need to make changes but one to keep in mind in future.

Reply:
Yeah, I'll have a think about how this could be implemented better. My first kneejerk reaction was partials like we've done elsewhere, but I don't think that's an elegant solution here.

Reply:
I think the alternative would be to construct the string in the backend and pass it straight through to the frontend as-is.

Reply:
(but to be clear, not suggesting you do this now)

Reply:
Agreed, I'm not a huge fan of the solution but I think it would be the correct one, given how many moving parts there are in the one string